core/src/Threads/Kokkos_ThreadsExec.cpp

```diff
@@ -78,6 +78,10 @@ void (*volatile s_current_function)(ThreadsExec &, const void *);
 const void *volatile s_current_function_arg = nullptr;
 
 struct Sentinel {
+#ifdef KOKKOS_COMPILER_INTEL
+  Sentinel() {}
+#endif
```

Comment:
Why are you not defaulting it?

Reply:
Because that would still trigger the warning.

Reply:
And why did you chose this over `static Sentinel const sentinel{};`

Reply:
I have not considered that so far, but it also seems to work.